Automattic/wp-calypso
# Reader Recent Feed: Refetch subscriptions from empty state

Related to https://github.com/Automattic/wp-calypso/pull/96988

## Proposed Changes

* Refetch subscriptions when the subscription count changes. This is needed to update from the empty state when onboarding is completed and subscriptions are added.
* Extract the subscription logic.

https://github.com/user-attachments/assets/4287dcdc-505a-4a6d-b7df-42f1e6c71817

## Why are these changes being made?


* Improve the Reader empty state experience.

## Testing Instructions



* Remove all of your subscriptions except to your own sites.
* Go to /read
* Click "Discover and subscribe to sites you'll love"
* Subscribe to one or more sites.
* Click Continue.
* The stream with the sites you've subscribed to should show.

## Pre-merge Checklist
